src/Commands/core/ImageFlushCommand.php

```diff
@@ -0,0 +1,80 @@
+
+use Drupal\image\Entity\ImageStyle;
+use Drush\Attributes as CLI;
+use Drush\Commands\AutowireTrait;
+use Drush\Utils\StringUtils;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\StyleInterface;
+
+#[AsCommand(
+    name: self::FLUSH,
+    description: 'Flush all derived images for a given style.',
+    aliases: ['if', 'image-flush']
+)]
+#[CLI\ValidateEntityLoad(entityType: 'image_style', argumentName: 'style_names')]
+#[CLI\ValidateModulesEnabled(modules: ['image'])]
+final class ImageFlushCommand extends Command
+{
+    use AutowireTrait;
+
+    const FLUSH = 'image:flush';
+
+    public function __construct(
+        private readonly LoggerInterface $logger,
+        private readonly StyleInterface $io
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addArgument('style_names', InputArgument::OPTIONAL, 'A comma delimited list of image style machine names. If not provided, user may choose from a list of names.')
+            ->addOption('all', null, InputOption::VALUE_NONE, 'Flush all derived images')
+            ->addUsage('image:flush thumbnail,large')
+            ->addUsage('image:flush --all')
+            ->setHelp('Immediately before running this command, web crawl your entire web site. Or use your Production PHPStorage dir for comparison.');
+    }
+
+    protected function interact(InputInterface $input, OutputInterface $output): void
+    {
+        $styles = array_keys(ImageStyle::loadMultiple());
+        $style_names = $input->getArgument('style_names');
+
+        if (empty($style_names) && !$input->getOption('all')) {
+            $styles_all = $styles;
+            array_unshift($styles_all, 'all');
+            $choices = array_combine($styles_all, $styles_all);
+            $style_names = $this->io->select(dt("Choose a style to flush"), $choices, 'all', scroll: 20);
+            if ($style_names == 'all') {
+                $style_names = implode(',', $styles);
+            }
+            $input->setArgument('style_names', $style_names);
+        }
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        // Needed for non-interactive requests.
+        if ($input->getOption('all')) {
+            $input->setArgument('style_names', implode(',', array_keys(ImageStyle::loadMultiple())));
+        }
+
+        foreach (ImageStyle::loadMultiple(StringUtils::csvToArray($input->getArgument('style_names'))) as $style_name => $style) {
+            $style->flush();
+            $this->logger->success(dt('Image style !style_name flushed', ['!style_name' => $style_name]));
```

Comment:
What's the purpose of `dt` function here?  The PSR-3 loggers must be capable of string interpolation. Also success method is not a part of  PSR-3. So the proper typehint for it should be `LoggerInterface&SuccessInterface`.

Reply:
Thanks for bringing up the typehint. When I use LoggerInterface&SuccessInterface, AutoWireTrait fails to find a service. I don't want to hint to DrushLoggerManager since Drupal will not want to inject that if it ever takes over our commands. We could potentially add more smarts to our AutoWireTrait. See https://www.drupal.org/project/drupal/issues/3417735 and https://symfony.com/blog/new-in-symfony-5-4-dependencyinjection-improvements#autowire-union-and-intersection-types for background. Thoughts? 

I removed dt() from the log messages - good idea. I have seen a styling issue with this approach (illegible text on a red background) when used with error() but we can deal with that when it comes up.

Reply:
Can we maybe just get rid of the success log level now?

Reply:
I think custom log levels is tricky thing. Though the PSR3 is a bit lenient about this.
```
Calling this method with a level not defined by this specification MUST throw a Psr\Log\InvalidArgumentException if the implementation does not know about the level. Users SHOULD NOT use a custom level without knowing for sure the current implementation supports it.
```

The alternative approach would by just passing special context to logger. So loggers that support this can handle this messages in a different way. For example, console logger could display it in green.
```php
$logger->info('Log message', ['success' => true]);
```

Reply:
> $logger->info('Log message', ['success' => true]);

That looks good to me. I think we have to use notice() or else it wont show up without -v. I would love to have a wrapper like $this->success() but we extend Command now and have nowhere to put that code.

Reply:
I think `$io` is more appropriate tool for such messages than logger.

Reply:
Yes, we can construct a Style object from $input and $output, which are parameters to execute. The logger is a little more fraught; we don't want people making their own logger from $output, but calling a command directly with a separate logger is not convenient with the logger passed via the DI container. I therefore think that moving folks away from the logger and to a Style method is preferable.